src/settings/ClosingReasonsForm.js

```diff
@@ -0,0 +1,169 @@
+
+
+import { Required } from '../components/Utils/Validate';
+
+class ClosingReasonsForm extends Component {
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    mutator: PropTypes.shape({
+      closingReasons: PropTypes.shape({
+        POST: PropTypes.func.isRequired,
+        PUT: PropTypes.func.isRequired,
+        DELETE: PropTypes.func.isRequired,
+      }),
+      recordId: PropTypes.shape({
+        replace: PropTypes.func,
+      }),
+    }).isRequired,
+    resources: PropTypes.shape({
+      closingReasons: PropTypes.shape({
+        records: PropTypes.arrayOf(PropTypes.object),
+      }),
+    }),
+    stripes: PropTypes.shape({
+      store: PropTypes.object.isRequired,
+    }),
+    dispatch: PropTypes.func.isRequired,
+    change: PropTypes.func.isRequired,
+  };
+
+  addFields = (fields) => fields.push({});
+
+  removeFields = (fields, index) => fields.remove(index);
+
+  onChangeInput = (e, propertyName) => {
+    const { dispatch, change } = this.props;
+
+    dispatch(change(propertyName, e));
+  };
+
+  renderForm = ({ fields }) => {
+    return (
+      <Row>
+        <Col xs={12}>
+          {fields.length === 0 &&
+            <Col xs={6}>
+              <div>
+                <em>
+                  <FormattedMessage id="ui-orders.settings.closingReasons.add" />
+                </em>
+              </div>
+            </Col>
+          }
+          {fields.map(this.renderSubForm)}
+        </Col>
+        <Col xs={12} style={{ paddingTop: '10px' }}>
+          <Button onClick={() => this.addFields(fields)}>
+            <FormattedMessage id="ui-orders.settings.closingReasons.addBtn" />
+          </Button>
+        </Col>
+      </Row>
+    );
+  };
+
+  renderSubForm = (elem, index, fields) => {
+    return (
+      <Row key={index}>
+        <Col xs={8}>
+          <Field
+            component={TextField}
+            fullWidth
+            label={<FormattedMessage id="ui-orders.settings.closingReasons.reason" />}
+            name={`${elem}.value`}
+            validate={[Required]}
+            onChange={e => this.onChangeInput(e.target.value, `${elem}.value`)}
+          />
+        </Col>
+        <Col xs={1} style={{ paddingTop: '4px' }}>
+          <br />
+          <Button onClick={() => this.onSaveClosingReason(index)}>
+            <FormattedMessage id="ui-orders.settings.closingReasons.saveBtn" />
+          </Button>
+        </Col>
+        <Col xs={1} style={{ paddingTop: '4px' }}>
+          <br />
+          <Button
+            buttonStyle="danger"
+            onClick={() => this.onRemoveClosingReason(fields, index)}
+          >
+            <FormattedMessage id="ui-orders.settings.closingReasons.removeBtn" />
+          </Button>
+        </Col>
+      </Row>
+    );
+  }
+
+  onSaveClosingReason = (index) => {
+    const { stripes: { store }, resources } = this.props;
+    const formValues = getFormValues('closing-reasons')(store.getState());
```

Comment:
you don't need this at all. I think you need to re-think your approach with sub-forms, they have submit button/callback, and they pass needed values to it. You can reference redux-form example https://redux-form.com/7.4.2/examples/fieldarrays/